tests/test_docker.py

```diff
@@ -88,6 +88,69 @@ def test_instance_activate_need_pull_image(agent, responses):
 
 
 @if_docker
+def test_docker_client_pull():
+    client = docker_client()
+    try:
+        client.remove_image(image='quay.io/wizardofmath/hellodocker:latest',
+                            force=True)
+    except APIError:
+        pass
+    container = client.pull(repository='quay.io/wizardofmath/hellodocker',
+                            tag='latest')
+    print(container)
+    assert container is not None
+
+
+def _need_pull_image_by_name(agent, responses, image_name):
+    _delete_container('/c861f990-4472-4fa1-960f-65171b544c28')
+
+    try:
+        docker_client().remove_image(image_name)
+    except APIError:
+        pass
+
+    def pre(req):
+        instance = req['data']['instanceHostMap']['instance']
+        for nic in instance['nics']:
+            nic['macAddress'] = ''
+        instance['data']['fields']['imageUuid'] = image_name
+
+    def post(req, resp):
+        responseInstance = resp['data']['instanceHostMap']['instance']['+data']
+        resp_img_uuid = responseInstance['dockerContainer']['Image']
+        parsed_name = parse_repo_tag(resp_img_uuid)
+        sent_parsed = parse_repo_tag(image_name)
+        assert parsed_name['repo'] == sent_parsed['repo']
+        if sent_parsed['tag'] != '':
+            assert parsed_name['tag'] == sent_parsed['tag'] \
+                   or parsed_name['tag'] != None
```

Comment:
What is the scenario where the first condition will fail (```parsed_name['tag'] == sent_parsed['tag']```)  and the second part will pass ( ```parsed_name['tag'] != None```)?